yarn-project/circuits.js/src/kernel/public_kernel.ts

```diff
@@ -12,7 +14,11 @@ import { callAsyncWasm } from '../utils/call_wasm.js';
  * @returns The public inputs.
  */
 export async function simulatePublicKernelCircuit(input: PublicKernelInputs): Promise<KernelCircuitPublicInputs> {
-  return callAsyncWasm(await CircuitsWasm.get(), 'public_kernel__sim', input, KernelCircuitPublicInputs);
+  const result = await publicKernelSim(await CircuitsWasm.get(), input);
+  if (result instanceof CircuitError) {
```

Comment:
I usually prefer type guard methods instead of `instanceof`. WDYT about switching to `isCircuitError` here, since we already have that method?

Reply:
I just prefer operating on raw interfaces when possible (I know, the whole mess of classes in circuits structs doesn't seem like that). But you're right that in here instanceof should get the job done, and even better, won't lead to issues if the object returned by the circuit has a `message` and a `code` field. So disregard my original comment!